src/microbe_stage/systems/MicrobeAISystem.cs

```diff
@@ -330,18 +332,30 @@ private void ChooseActions(in Entity entity, ref MicrobeAI ai, ref CompoundAbsor
             control.SetMucocystState(ref organelles, ref compoundStorage, entity, false);
         }
 
+        float atpLevel = compounds.GetCompoundAmount(Compound.ATP);
+
         // If this microbe is out of ATP, pick an amount of time to rest
-        if (compounds.GetCompoundAmount(Compound.ATP) < 1.0f)
+        if (atpLevel < 1.0f)
         {
             // Keep the maximum at 95% full, as there is flickering when near full
             ai.ATPThreshold = 0.95f * speciesFocus / Constants.MAX_SPECIES_FOCUS;
         }
 
         if (ai.ATPThreshold > MathUtils.EPSILON)
         {
-            if (compounds.GetCompoundAmount(Compound.ATP) <
-                compounds.GetCapacityForCompound(Compound.ATP) * ai.ATPThreshold)
+            if (atpLevel < compounds.GetCapacityForCompound(Compound.ATP) * ai.ATPThreshold)
             {
+                if (cellHealth.CurrentHealth > 2 * Constants.ENGULF_NO_ATP_DAMAGE)
+                {
+                    // Even if we are out of ATP and there is microbe nearby, engulf them.
+                    // make sure engulfing doesn't kill the cell
+                    if (CheckForHuntingConditions(ref ai, ref position, ref organelles, ref ourSpecies, ref engulfer,
```

Comment:
Maybe it would be a good idea to move this check to after the `if (!outOfSomething) { ... }` code block? So that if the microbe has all the necessary compounds, but still lacks ATP because of some temporary conditions (like sprinting) it just waits to generate some more ATP instead of hunting